src/components/ItemList.js

```diff
@@ -1,41 +1,67 @@
-import { collection, query, where, onSnapshot } from 'firebase/firestore';
-import { useEffect, useState } from 'react';
-import { db } from '../lib/firebase.js';
+import { useState } from 'react';
+import useFirebaseSnapshot from '../hooks/useFirebaseSnapshot.js';
+import cleanData from '../utils/cleanData.js';
 import { Link } from 'react-router-dom';
 
 const ItemList = () => {
-  const [docs, setDocs] = useState([]);
-
-  useEffect(() => {
-    const token = localStorage.getItem('token');
-    const q = query(
-      collection(db, 'shopping-list'),
-      where('token', '==', token),
-    );
-    const unsubscribe = onSnapshot(q, (querySnapshot) => {
-      const items = [];
-      querySnapshot.forEach((doc) => {
-        items.push(doc.data());
-      });
-      setDocs(items);
+  const [searchInput, setSearchInput] = useState('');
+  const [filteredResults, setFilteredResults] = useState('');
+
+  const { docs, loading } = useFirebaseSnapshot();
```

Comment:
I'm curious what prompted you to add the loading functionality. Just wanted to provide a better user experience while the items are being read from the database?

Reply:
yes, you're correct.